lightning/src/ln/channel.rs

```diff
@@ -3553,9 +3665,28 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			return Err(ChannelError::Ignore("Cannot send value that would put us over the max HTLC value in flight our peer will accept"));
 		}
 
+		if !self.channel_outbound {
+			// Check that we won't violate the remote channel reserve by adding this HTLC.
+
+			let remote_balance_msat = self.channel_value_satoshis * 1000 - self.value_to_self_msat;
+			let remote_chan_reserve_msat = Channel::<ChanSigner>::get_remote_channel_reserve_satoshis(self.channel_value_satoshis);
+			// 1 additional HTLC corresponding to this HTLC.
+			let remote_commit_tx_fee_msat = self.next_remote_commit_tx_fee_msat(1);
+			if remote_balance_msat < remote_chan_reserve_msat + remote_commit_tx_fee_msat {
+				return Err(ChannelError::Ignore("Cannot send value that would put them under remote channel reserve value"));
+			}
+		}
+
+		let chan_reserve_msat = self.local_channel_reserve_satoshis * 1000;
+		// The +1 is for the HTLC currently being added to the commitment tx and
+		// the +2 is for the fee spike buffer.
+		let local_commit_tx_fee_msat = if self.channel_outbound {
+			self.next_local_commit_tx_fee_msat(1 + 2)
+		} else { 0 };
```

Comment:
Ok, I read the spec again and can see how there may be some ambiguity around the fee spike buffer. The spec states:

> SHOULD NOT offer `amount_msat` if, after adding that HTLC to its commitment transaction, its remaining balance doesn't allow it to pay the commitment transaction fee when receiving or sending a future additional non-dust HTLC while maintaining its channel reserve. It is recommended that this "fee spike buffer" can handle twice the current `feerate_per_kw` to ensure predictability between implementations.

I thought that "twice the current `feerate_per_kw`" should apply to fees for the entire commitment transaction, including the HTLC sent here and a "future additional non-dust HTLC".

However, here we are adding _two_ "future additional non-dust HTLC"s as a way to handle "twice the current `feerate_per_kw`". i.e, only applying the fee rate spike to the future HTLC.

I wasn't sure what the spec's intention was, so I dug into the PR adding it. I found https://github.com/lightningnetwork/lightning-rfc/pull/740#discussion_r399740407 which shows there was similar confusion. From my reading of the discussion, the 2x multiplier should apply to fees for the entire commitment transaction, not just the future HTLC. Let me know if I'm misreading that discussion.

In other words, I think our code should be:

```rust
2 * self.next_local_commit_tx_fee_msat(1 + 1)
```

Or possibly move the multiplication inside `next_local_commit_tx_fee_msat`.

Reply:
Oof, I see what you mean. I think you're right. @TheBlueMatt I'm gonna move forward with this change

Reply:
Updated in 0ed82f8fec1ca6e2418d4097c86ac71e8ba2985c